File: backend/KnowledgeSharingPlatform.Api/Models/Resource.cs

namespace KnowledgeSharingPlatform.Api.Models
{
    public class Resource
    {
        public int resource_id { get; set; }
        public int event_id { get; set; }
        public string file_type { get; set; } = string.Empty;
        public string file_path { get; set; } = string.Empty;
        //public bool IsActive { get; set; }
        public int ViewCount { get; set; }
        public DateTime uploaded_at { get; set; }
        public string status { get; set; }
    }
}

File: Frontend/ksp/src/app/model/resource.ts
// src/app/models/event.model.ts
export interface Event {
  event_id: number;
  title: string;
  type: string;
  startDate: string;     // ISO string; can be Date if parsed
  endDate: string;
  organized_By: string;
  keywords: string;
  userId: number;
  status: string;
}



export interface Resource {
  resource_Id: number;
  event_Id: number;
  file_Type: string;
  file_Path: string;
  views: number;
  uploaded_at: string;
  status: string;
   event: Event; 
}
